lib/helper.js

```diff
@@ -24,6 +24,10 @@ function encodeKey(metricName, tags) {
   }
   // Sort the tags based on their keys to ensure we don't duplicate metrics if the same tags come out of order.
   var tagsArray = Object.keys(tags).sort().map(key => [key, tags[key]]);
+  var tagsType = typeof tags;
```

Comment:
Couple of things here:
1. These checks should be done before tagsArray is executed.
2. typeof and types in general are very tricky in javascript.  You should try to handle as many cases as possible.  Please see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Examples for the sort of weirdness that can occur.
3. prefer !== over != in javascript
4. avoid throwing non-Errors.  This is easily fixed by switching to `throw new Error("message here")`